src/display.py

```diff
@@ -0,0 +1,62 @@
+'''
+seven segment diagnostic display on the data lines
+
+-cannot use inside a pinball machine
+-intended for network, memeory tests etc..  not machine testing
+
+'''
+from machine import Pin
+import time
+
+pins = []
+
+segment_map = {
+    '0': 0b00111111, '1': 0b00000110, '2': 0b01011011, '3': 0b01001111,
+    '4': 0b01100110, '5': 0b01101101, '6': 0b01111101, '7': 0b00000111,
+    '8': 0b01111111, '9': 0b01101111, 'A': 0b01110111, 'B': 0b01111100,
+    'C': 0b00111001, 'D': 0b01011110, 'E': 0b01111001, 'F': 0b01110001,
+    'G': 0b00111101, 'H': 0b01110110, 'I': 0b00000110, 'J': 0b00011110,
+    'L': 0b00111000, 'M': 0b01001111, 'N': 0b01010100, 'O': 0b01011100,
+    'P': 0b01110011, 'Q': 0b01100111, 'R': 0b01010000, 'S': 0b01101101, 
+    'T': 0b01111000, 'U': 0b00111110, 'X': 0b01110110, 'V': 0b00111110, 
+    'Y': 0b01101110, '.': 0b10000000, ' ': 0b00000000
+}
+
+
+def initialize():
+    #turn off PIO state machine?
+    #  should already be off with the board not in a machine (fault detected)
+
```

Comment:
FYI I don't get a fault detected when not installed in a machine.

Reply:
We need to check your hardware.  Possible the old boards don't do this correctly or it is turned off in the dev version of main.py   The new main with version 4 or 5 boards should get the fault on power up.

Reply:
oh, I might be wrong.  I forgot we now look at bus activity when there should not be any.  And an idle board on the bench will actually pass this test.  Of course - on the bench there is no harm

Reply:
I can test on v5 hardware sometime tomorrow, I don't have my board flashed with micropython yet.

Reply:
I found a bug that was causing the non-error and resolved it.